### PR TITLE
Fix: do not override the workload name if its specified

### LINKED ISSUE #2312: [Feature] overwrite the workload name with component name only if not defined
**Is your feature request related to a problem? Please describe.**
In some situation, I wish to use the same name for multiple components of different type in one application, e.g. the application name.

However, it's required to set different names for these components, and the name of the workload is set, by kubevela automatically, with the name of the component.
```
//// @assembly.go#assembleWorkload
// use component name as workload name
// override the name set in render phase if exist
wl.SetName(compName)
```

**Describe the solution you'd like**
Use component name as workload name only if the name of workload is not defined.
```
//// @assembly.go#assembleWorkload
// use component name as workload name only if not defined
if len(wl.GetName()) == 0 {
    wl.SetName(compName)
}
```

**Describe alternatives you've considered**
It's not forward compatible for those applications that overwrite the workload name with a different component name.

**Additional context**

### PR DESCRIPTION
**Pull request by FogDong**
**What this PR does / why we need it**:
do not override the workload name if its specified

**Which issue(s) this PR fixes**:

Fixes #2312

**Special notes for your reviewer**:
/cc @wangyikewxgm

### CODE REVIEW

File: pkg/appfile/appfile.go [hunk 1]
```diff
@@ -350,9 +350,11 @@ func (af *Appfile) setNamespace(obj *unstructured.Unstructured) {
 }
 
 func (af *Appfile) assembleWorkload(wl *unstructured.Unstructured, compName string, labels map[string]string) {
-	// use component name as workload name
+	// use component name as workload name if workload name is not specified
 	// override the name set in render phase if exist
```
Comment (wonderflow) [inline pkg/appfile/appfile.go:354]:
```suggestion
	// don't override the name set in render phase if exist
```

File: pkg/controller/core.oam.dev/v1alpha2/application/assemble/assemble.go [hunk 2]
```diff
@@ -292,9 +292,11 @@ func (am *AppManifests) validate() error {
 
 func assembleWorkload(compName string, wl *unstructured.Unstructured,
 	labels map[string]string, resources []*unstructured.Unstructured, appRev *v1beta1.ApplicationRevision, wop []WorkloadOption) (*unstructured.Unstructured, error) {
-	// use component name as workload name
+	// use component name as workload name if workload name is not specified
 	// override the name set in render phase if exist
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/application/assemble/assemble.go:296]:
```suggestion
	// don't override the name set in render phase if exist
```

**Review by wonderflow — APPROVED** [review]
ping @wangyikewxgm

**Review by leejanee — APPROVED** [review]

**Comment by wangyikewxgm** [comment]
最好有一个controller test完整走一遍reconcile之后，确保出来的workload的name是正确的

**Review by wangyikewxgm — APPROVED** [review]
最好有一个controller test完整走一遍reconcile之后，确保出来的workload的name是正确的

**Pull request merged**